app/test/unit/promise-with-timeout-test.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from 'chai'
+
+import { promiseWithMinimumTimeout } from '../../src/lib/promise'
+
+jest.useFakeTimers()
+
+describe('promiseWithMinimumTimeout', () => {
+  it('handles promise finishing before timeout', async () => {
+    const fastPromise = new Promise<number>((resolve, reject) => {
+      window.setTimeout(() => {
+        resolve(42)
+      }, 100)
+    })
+
+    const promise = promiseWithMinimumTimeout(() => fastPromise, 500)
+
+    // promise completes
+    jest.advanceTimersByTime(250)
```

Comment:
should we check that `promise` isn't resolved right after this?

Reply:
hmmm fair point. i think we could get what we want with a jest function mock

```js
describe('promiseWithMinimumTimeout', () => {
  it('handles promise finishing before timeout', async () => {
    const resolveWrapper = jest.fn().mockImplementation((resolve) => resolve(42))
    const fastPromise = new Promise<number>((resolve, reject) => {
      window.setTimeout(resolveWrapper(resolve), 100)
    })
     const promise = promiseWithMinimumTimeout(() => fastPromise, 500)
     // promise completes
    jest.advanceTimersByTime(250)
    // this would be a little cleaner with jest expect instead of chai expect
    expect(resolveWrapper.calls.length).equals(0)
...
```

this would be a little cleaner with jest instead of chai expect, but yeah.

Reply:
@outofambit maybe? let me try out how it feels

Reply:
Try out 16d996dd5 and see how it feels

Reply:
🎸rockin'!